pytuya/__init__.py

```diff
@@ -347,23 +362,39 @@ def set_value(self, index, value):
 
         payload = self.generate_payload(SET, {
             index: value})
-        
+
         data = self._send_receive(payload)
-        
+
         return data
-    
+
     def turn_on(self, switch=1):
         """Turn the device on"""
-        self.set_status(True, switch)
+        if self.set_status(True, switch) == None:
+            for i in range(0, 3):
+                if self.set_status(True, switch) == None:
+                    continue
+                else:
+                    return True
+            return False
+        else:
+            return True
 
     def turn_off(self, switch=1):
         """Turn the device off"""
-        self.set_status(False, switch)
+        if self.set_status(False, switch) == None:
+            for i in range(0, 3):
+                if self.set_status(False, switch) == None:
+                    continue
+                else:
+                    return True
+            return False
+        else:
+            return True
```

Comment:
looks like retry logic, should use retry count

Reply:
Again correctly